src/dsst_defacing_wf.py

```diff
@@ -58,153 +60,113 @@ def get_sess_dirs(subj_dir_path, mapping_dict):
-
-                copy_over_sidecar(nii_filepath, input_dir / anat_dir.relative_to(defacing_outputs), anat_dir)
-
-        # move QC images and afni intermediate files to a new directory
-        intermediate_files_dir = anat_dir / 'workdir'
-        intermediate_files_dir.mkdir(parents=True, exist_ok=True)
-        for dirpath in anat_dir.glob('*'):
-            if dirpath.name.startswith('workdir') or dirpath.name.endswith('QC'):
-                shutil.move(dirpath, intermediate_files_dir)
-
-        # if not no_clean:
-        #     shutil.rmtree(intermediate_files_dir)
-
-
-def generate_3d_renders(defaced_img, render_outdir):
-    rotations = [(45, 5, 10), (-45, 5, 10)]
-    for idx, rot in enumerate(rotations):
-        yaw, pitch, roll = rot[0], rot[1], rot[2]
-        outfile = render_outdir.joinpath('defaced_render_' + str(idx) + '.png')
-        fsleyes_render_cmd = f"fsleyes render --scene 3d -rot {yaw} {pitch} {roll} --outfile {outfile} {defaced_img} -dr 20 250 -in spline -bf 0.3 -r 100 -ns 500"
-        print(fsleyes_render_cmd)
-        run_command(fsleyes_render_cmd)
-
-
-def create_defacing_id_list(qc_dir):
-    rel_paths_to_orig = [re.sub('/orig.nii.gz', '', str(o.relative_to(qc_dir))) for o in qc_dir.rglob('orig.nii.gz')]
-    with open(qc_dir / 'defacing_id_list.txt', 'w') as f:
-        f.write('\n'.join(rel_paths_to_orig))
-
-
-def vqcdeface_prep(input_dir, defacing_output_dir):
-    defacing_qc_dir = defacing_output_dir.parent / 'QC_prep' / 'defacing_QC'
-    interested_files = [f for f in defacing_output_dir.rglob('*.nii.gz') if
-                        'workdir' not in str(f).split('/')]
-    for defaced_img in interested_files:
-        entities = defaced_img.name.split('.')[0].split('_')
-        vqcd_subj_dir = defacing_qc_dir / f"{'/'.join(entities)}"
-        vqcd_subj_dir.mkdir(parents=True, exist_ok=True)
-
-        defaced_link = vqcd_subj_dir / 'defaced.nii.gz'
-        if not defaced_link.exists():
-            defaced_link.symlink_to(defaced_img)
-            generate_3d_renders(defaced_img, vqcd_subj_dir)
-
-        img = list(input_dir.rglob(defaced_img.name))[0]
-        img_link = vqcd_subj_dir / 'orig.nii.gz'
-        if not img_link.exists(): img_link.symlink_to(img)
+def main():
+    # get command line arguments
+    args = get_args()
 
-    create_defacing_id_list(defacing_qc_dir)
+    input_dir = Path(args.bids_dir).resolve()
+    output = Path(args.output_dir).resolve()
+    no_clean = args.no_clean
 
-    vqcdeface_cmd = f"vqcdeface -u {defacing_qc_dir} -i {defacing_qc_dir / 'defacing_id_list.txt'} -m orig.nii.gz -d defaced.nii.gz -r defaced_render"
+    to_deface = []
+    # only for one subject (and all their sessions, if present)
+    if args.participant_label:
+        to_deface = glob(os.path.join(args.bids_dir, args.participant_label, "ses-*"))
+        if to_deface == []:
+            to_deface = glob(os.path.join(args.bids_dir, args.participant_label))
```

Comment:
I'm back to testing a bit more thoroughly with all possible combination of options and so, when I ran the following command I got an error:
Command: `python src/dsst_defacing_wf.py ../defacing-test-dataset/ ../testing-pipeline-pn -p 01090 -n 4`
Error: 
```bash
"(dsstdeface)[arshithab@cn2877 dsst-defacing-pipeline]$ python src/dsst_defacing_wf.py ../defacing-test-dataset/ ../testing-pipeline-pn -p 01090 -n 4
['01090']
../defacing-test-dataset
Traceback (most recent call last):
  File ""/gpfs/gsfs10/users/arshithab/dsst-defacing-pipeline/src/dsst_defacing_wf.py"", line 174, in <module>
    main()
  File ""/gpfs/gsfs10/users/arshithab/dsst-defacing-pipeline/src/dsst_defacing_wf.py"", line 75, in main
    to_deface = glob(os.path.join(args.bids_dir, args.participant_label, ""ses-*""))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ""<frozen posixpath>"", line 90, in join
  File ""<frozen genericpath>"", line 152, in _check_arg_types
TypeError: join() argument must be str, bytes, or os.PathLike object, not 'list'"
```

To fix this and other issues related to the if-else conditional block, I changed lines 72 to 90 as follows:
```suggestion
    # for one subject or list of subjects (and all their sessions, if present)
    if args.participant_label:
        for p in args.participant_label:
            to_deface.extend(glob(os.path.join(args.bids_dir, f'sub-{p}', "ses-*")))
            if not to_deface:
                to_deface = glob(os.path.join(args.bids_dir, f'sub-{p}'))

    # only for one subset of sessions
    if args.session_id:
        for s in args.session_id:
            to_deface = glob(os.path.join(args.bids_dir, "sub-*", f'ses-{s}'))
    # for all sessions
    if not (args.participant_label or args.session_id):
        session_check = glob(os.path.join(args.bids_dir, "sub-*", "ses-*"))
        if session_check:
            to_deface = session_check

        # for all subjects
        else:
            to_deface = glob(os.path.join(args.bids_dir, "sub-*"))
```